### PR TITLE
Add service + allow control over what is autosaved

### PR DESCRIPTION
**Pull request by akonwi**
This is a step towards resolving #43. My Git-Plus package observes commit message files and if a save is triggered before the message is finished the commit happens anyway

### CODE REVIEW

**Comment by akonwi** [comment]
Has anyone looked at this?

**Comment by jsg2021** [comment]
I'm looking forward to this.

**Comment by akonwi** [comment]
@50Wliu Just following up. Does this have a chance at being accepted?

File: lib/autosave.coffee [hunk 1]
```diff
@@ -9,6 +10,8 @@ module.exports =
 
   subscriptions: null
 
+  provideControls: -> dontSaveIf
```
Comment (nathansobo) [inline lib/autosave.coffee]: What do you think about providing an object with a method so we can add other methods without breaking major version compatibility with the existing service?

File: spec/controls-spec.coffee [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Controls = require '../lib/controls'
+
+describe 'Controls', ->
```
Comment (nathansobo) [inline spec/controls-spec.coffee]: I'd also like to see a more integrated test of `autosave` itself that ensures it interacts with the service and doesn't save if a predicate returns true.

File: README.md [hunk 3]
```diff
@@ -6,3 +6,22 @@ closed.
 This package is disabled by default and can be enabled via the
 `autosave.enabled` config setting or from the Autosave section of the Settings
 view (`cmd-,`).
+
+## Service API
+
+### package.json
```
Comment (nathansobo) [inline README.md]: This example is great, but could you add some text in english describing this as well?

File: README.md [hunk 4]
```diff
@@ -6,3 +6,22 @@ closed.
...
+    "versions": {
+      "0.22.0": "consumeAutosave"
+    }
+  }
+}
+```
+
+### package initialize
+``` coffeescript
+consumeAutosave: (dontSaveIf) ->
```
Comment (nathansobo) [inline README.md]: Let's use ES6 for this example instead because we're trying to phase out CoffeeScript, though it's going to take a while.

**Review by nathansobo — CHANGES_REQUESTED** [review]
Thanks! This looks like a cool addition. If you can address my requested changes I think we're close. Perhaps @50Wliu could help with the documentation description if you need input there.

**Comment by akonwi** [comment]
Great! I'll make the changes

**Comment by akonwi** [comment]
I removed the previously written tests for `Controls` and replaced them with an integrated test using a predicate. The description in the README may need more tweaking though

**Review by nathansobo — APPROVED** [review]

**Comment by nathansobo** [comment]
@akonwi I made a few adjustments I'd like to push to your branch... can you [follow these instructions](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) to enable me to push to your fork for this branch?

**Comment by akonwi** [comment]
@nathansobo Done

**Comment by akonwi** [comment]
Cool! I just made a couple more changes for cleanup

**Comment by nathansobo** [comment]
Great additions! Thanks for catching both of those things. Really appreciate you working with us on this one despite the extremely long delay in getting an initial review. Thanks for your efforts!

**Pull request merged**